src/libopenrave/kinbody.cpp

```diff
@@ -346,25 +346,21 @@ void KinBody::KinBodyInfo::DeserializeJSON(const rapidjson::Value& value, dReal
 
     if (value.HasMember("links")) {
         _vLinkInfos.reserve(value["links"].Size() + _vLinkInfos.size());
-        for (rapidjson::Value::ConstValueIterator it = value["links"].Begin(); it != value["links"].End(); ++it) {
-            UpdateOrCreateInfoWithNameCheck(*it, _vLinkInfos, "name", fUnitScale, options);
-        }
+        UpdateOrCreateInfosWithNameCheck(value["links"].Begin(), value["links"].End(), _vLinkInfos, "name", fUnitScale, options);
 
-        // if has conflicting names, should error here
-        for(int ilink0 = 0; ilink0 < (int)_vLinkInfos.size(); ++ilink0 ) {
-            for(int ilink1 = ilink0+1; ilink1 < (int)_vLinkInfos.size(); ++ilink1 ) {
-                if( _vLinkInfos[ilink0]->_name == _vLinkInfos[ilink1]->_name ) {
-                    throw OPENRAVE_EXCEPTION_FORMAT("Body '%s' has info with link[%d] and link[%d] having the same linkname '%s', which is not allowed. link[%d].id='%s', link[%d].id='%s'", _name%ilink0%ilink1%_vLinkInfos[ilink0]->_name%ilink0%_vLinkInfos[ilink0]->_id%ilink1%_vLinkInfos[ilink1]->_id, ORE_Assert);
-                }
+        // Assert that we don't have any duplicate link names
```

Comment:
@rschlaikjer 

might be minor comment, but does `UpdateOrCreateInfosWithNameCheck` guarantee the uniqueness of the link names?

If so, do we need the following additional checking?

Reply:
It's still possible to have duplicate names - since we match on id preferentially to name, the following would create two entries in `UpdateOrCreateInfosWithNameCheck` (IDs are different) but still have a name collision:
```json
[ {"id": "a", "name": "n"}, {"id": "b", "name": "n"}]
```

Reply:
i see~